com.chartboost.helium/Android~/helium-android-bridge/src/main/kotlin/com/chartboost/heliumsdk/unity/HeliumEventProcessor.kt

```diff
@@ -0,0 +1,95 @@
+package com.chartboost.heliumsdk.unity
+
+import android.util.Log
+import com.chartboost.heliumsdk.ad.HeliumAdError
+import org.json.JSONException
+import org.json.JSONObject
+
+object HeliumEventProcessor {
+    private const val TAG = "HeliumEventProcessor"
+    private const val EMPTY_STRING = ""
+
+    @JvmStatic
+    fun serializeHeliumEvent(
+        placement: String?,
+        heliumAdError: HeliumAdError?,
+        eventConsumer: HeliumEventConsumer<String?, Int, String?>
+    ) {
+        var placementName: String? = EMPTY_STRING
```

Comment:
welcome to kotlin! Null safety is built into the language! This method would make a lot more sense if placement was not null, but we can make do.
However, your event consumer takes a nullable placement name, so you don't need to do this.
`eventConsumer.acccept(placement, heliumAdError?.getCode() ?: -1, heliumAdError?.getMessage())` is perfectly fine. If you'd like them to be all not null, you can always do
`eventConsumer.accept(placement ?: "", heliumAdError?.getCode() ?: -1, heliumAdError?.getMessage() ?: "")`

another thing to note is that getCode() and getMessage() look like properties. if there is a `code` and `message` property, you can just use `heliumAdError?.code`

Reply:
Thanks for the clarification, it was my first time getting into kotlin so I wanted to get first hand feedback from the conversion and see what input was available, I have not understood this and will keep it in mind for the future, the changes were added as well.